consensus/src/simplex/actors/voter/state.rs

```diff
@@ -643,62 +653,47 @@ impl<E: Clock + Rng + CryptoRng + Metrics, S: Scheme, D: Digest> State<E, S, D>
     /// Finds the parent payload for a given view by walking backwards through
     /// the chain, skipping nullified views until finding a certified payload.
     fn find_parent(&self, view: View) -> Result<(View, D), View> {
-        if view == GENESIS_VIEW {
-            return Ok((GENESIS_VIEW, self.genesis.unwrap()));
-        }
-        let mut cursor = view - 1;
+        // If the view is the genesis view, consider it to be its own parent.
+        let mut cursor = view.saturating_sub(1);
         loop {
-            if cursor == GENESIS_VIEW {
-                return Ok((GENESIS_VIEW, self.genesis.unwrap()));
-            }
+            // Return the first notarized or finalized parent.
             if let Some(parent) = self.quorum_payload(cursor) {
                 return Ok((cursor, *parent));
             }
-            if self.is_nullified(cursor) {
-                if cursor == GENESIS_VIEW {
-                    return Ok((GENESIS_VIEW, self.genesis.unwrap()));
-                }
-                cursor -= 1;
-                continue;
+
+            // If the view is also not nullified, there is a gap in certificates.
+            if !self.is_nullified(cursor) {
+                return Err(cursor);
             }
-            return Err(cursor);
+
+            cursor = cursor.checked_sub(1).expect("cursor should not wrap");
         }
     }
 
-    /// Returns the payload of the parent's certificate, ensuring it sits between the last
-    /// finalized view and the current view while every skipped view is nullified.
-    fn parent_payload(&self, current_view: View, proposal: &Proposal<D>) -> Option<D> {
```

Comment:
removed `current_view` as a parameter since it should always be equal to `proposal.view()`. Another way to do this is to just pass in `current_view` and `parent_view` (not passing in the reference to the entire proposal)